pkg/registry/common/refresh/nse_registry_client.go

```diff
@@ -20,23 +20,45 @@ import (
 	"context"
 	"time"
 
-	"github.com/golang/protobuf/ptypes/timestamp"
-
 	"github.com/golang/protobuf/ptypes/empty"
-	"github.com/networkservicemesh/api/pkg/api/registry"
 	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
 
 	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/logger/logruslogger"
 )
 
 type refreshNSEClient struct {
 	chainContext          context.Context
 	nseCancels            cancelsMap
-	retryDelay            time.Duration
 	defaultExpiryDuration time.Duration
 }
 
-func (c *refreshNSEClient) startRefresh(ctx context.Context, client registry.NetworkServiceEndpointRegistryClient, nse *registry.NetworkServiceEndpoint) {
+// NewNetworkServiceEndpointRegistryClient creates new NetworkServiceEndpointRegistryClient that will refresh expiration
+// time for registered NSEs
+func NewNetworkServiceEndpointRegistryClient(options ...Option) registry.NetworkServiceEndpointRegistryClient {
+	c := &refreshNSEClient{
+		defaultExpiryDuration: time.Minute * 30,
+		chainContext:          context.Background(),
+	}
+
+	for _, o := range options {
+		o.apply(c)
+	}
+
+	return c
+}
+
+func (c *refreshNSEClient) startRefresh(
+	ctx context.Context,
+	client registry.NetworkServiceEndpointRegistryClient,
+	nse *registry.NetworkServiceEndpoint,
+) {
+	_, logEntry := logruslogger.New(ctx)
```

Comment:
Please use logger from passed context

Reply:
It is a chain context, there can be no logger:
```go
ctx, cancel := context.WithCancel(c.chainContext)
c.startRefresh(ctx, nextClient, refreshNSE)
```

Reply:
Note: logger will be placed in cmd applications.

Reply:
See here https://github.com/networkservicemesh/cmd-nsmgr/blob/master/main.go#L46

Reply:
Yep, fixed.

Reply:
Now default chain context is:
```go
logger.WithLog(context.Background())
```